### ISSUE TITLE
TextureThumbnailSource Doesn't work on curtain items.

### ISSUE THREAD

**Issue by candypheonixIO** [Frooxius/NeosPublic#761]
I've tried saving my latest youtube player with a thumbnail source, even tried giving it a different parent, and every time I saved it with the TextureThumbnailSource component, it would not display the custom thumbnail in the inventory.

Here is the texture and component in the inspector.
![2020-07-10 15 34 51](https://user-images.githubusercontent.com/50785867/87192192-fb5feb00-c2c3-11ea-9d43-bf064afcd7f0.jpg)

And this is what it looks like in the inventory.
![20200710153239_1](https://user-images.githubusercontent.com/50785867/87192275-22b6b800-c2c4-11ea-8452-ade23c405430.jpg)

**Comment by Frooxius** [comment]
Where is the TextureThumbnailSource placed in relation to the root of the item you're saving? Do you have multiple instances by any chance?

For this component to work it needs to be right at the root and there only needs to be one.

**Comment by candypheonixIO** [comment]
The first image shows the only instance of it

**Comment by Frooxius** [comment]
What about my other question?

If it fulfills those criteria, I'll need some way to replicate the problem. Can you either send me a copy of this in-game or provide steps to replicate this?

**Comment by candypheonixIO** [comment]
There was no other question, and the item can be found in my public folder [PUBLIC] VRtualis

You can use that to replicate the issue

**Comment by Frooxius** [comment]
I suggest you reread my first post. I wouldn't be asking you about the other question if I didn't ask one.

**Comment by candypheonixIO** [comment]
I should have already answered it; the first image shows the only instance, and you can also see that i'm selecting the root, and only root that is being saved.

**Comment by Frooxius** [comment]
Thanks for the answer. Unfortunately it's not clear from the image alone and I'd rather not make assumptions before investigating issue. I'll have a look and see what I can find out.

**Comment by candypheonixIO** [comment]
alrighty

**Comment by Frooxius** [comment]
I've checked, but I can't find the proper version in the inventory. Where exactly is it located? I can only find 0.2.31 in the WIP Items / YouTube Player and that one doesn't have a thumbnail on it at all.

**Comment by candypheonixIO** [comment]
Its on the first page, nit in any folders

**Comment by Frooxius** [comment]
I found it thanks. I checked the item. You actually do have multiple instances of TextureThumbnailSource in the hierarchy (a lot of them seem to be for Blueprints and other stuff) so it's triggering the heuristics to avoid using them. You'll need remove all of them and it'll work.

I've added a button "Ensure single thumbnail source" to the component which makes this process much easier in 2020.7.11.1366. Simply click it in the thumbnail source at the root.
